### PR TITLE
Issue 303

### LINKED ISSUE #303: Dockerize improvements
@Capprin : comments for improvements to #288 :

example url dates should be YYYY-MM-DD
default magnitudes should be null (no min/max)

buildProductSearchSql
- should this use currentProducts by default?

curly braces always:
https://github.com/usgs/earthquake-event-ws/pull/288/files#diff-9551ba515f29f78dbbc526e59123fc39R1851

could this be if `$query->isDeleted === false` ?
https://github.com/usgs/earthquake-event-ws/pull/288/files#diff-9551ba515f29f78dbbc526e59123fc39R1862

similarly, could this use currentProducts when !isDeleted and !isSuperseded?
https://github.com/usgs/earthquake-event-ws/pull/288/files#diff-9551ba515f29f78dbbc526e59123fc39R1872

exit is not needed after a throw:
https://github.com/usgs/earthquake-event-ws/pull/288/files#diff-9551ba515f29f78dbbc526e59123fc39R2033

### PR DESCRIPTION
**Pull request by Capprin**
closes #303

### CODE REVIEW

File: src/lib/classes/pdl/ProductIndex.class.php [hunk 1]
```diff
@@ -1723,7 +1723,7 @@ private function buildProductSearchSql($query) {
     );
 
     //get table
-    $sql .= sprintf(" FROM %s ps ",self::SUMMARY_TABLE);
+    $sql .= sprintf(" FROM %s ps ", isset($query->includeSuperseded)?self::SUMMARY_TABLE:self::CURRENT_TABLE);
```
Comment (jmfee-usgs) [inline src/lib/classes/pdl/ProductIndex.class.php:1726]: This should just be `$query->includeSuperseded ? ...`.  The value is always set, but defaults to false.

Also use spaces around the ternary operators `?` and `:` for readability.

File: src/lib/classes/pdl/ProductIndex.class.php [hunk 2]
```diff
@@ -1848,46 +1848,28 @@ private function buildProductSearchSql($query) {
...
       $params[] = $query->minLongitude;
     }
 
     //deleted
-    if (isset($query->includeDeleted)) {
-      if (!($query->includeDeleted)) {
-        $where[] = 'ps.' . self::SUMMARY_STATUS . '<>?';
-        $params[] = 'DELETE';
-      }
+    if ($query->includeDeleted === false) {
```
Comment (jmfee-usgs) [inline src/lib/classes/pdl/ProductIndex.class.php:1866]: This is easier to read as

`if (!$query->includeDeleted) {`

File: src/lib/classes/pdl/ProductIndex.class.php (hunk 1 above)
Comment (Capprin) [reply src/lib/classes/pdl/ProductIndex.class.php]: to my eye, the value isn't set; we have to check every parameter for its existence. still, updated for readability.

File: src/lib/classes/pdl/ProductIndex.class.php (hunk 2 above)
Comment (Capprin) [reply src/lib/classes/pdl/ProductIndex.class.php]: included isset, and rewrote as the above for readability

File: src/lib/classes/pdl/ProductIndex.class.php [hunk 3]
```diff
@@ -1723,7 +1723,7 @@ private function buildProductSearchSql($query) {
     );
 
     //get table
-    $sql .= sprintf(" FROM %s ps ",self::SUMMARY_TABLE);
+    $sql .= sprintf(" FROM %s ps ", isset($query->includeSuperseded) && $query->includeSuperseded ? self::SUMMARY_TABLE : self::CURRENT_TABLE);
```
Comment (jmfee-usgs) [inline src/lib/classes/pdl/ProductIndex.class.php:1726]: the `isset` check isn't needed (it's always set).  That's more important in an associative array where the key may or may not exist.

File: src/lib/classes/pdl/ProductIndex.class.php [hunk 4]
```diff
@@ -1848,46 +1848,28 @@ private function buildProductSearchSql($query) {
...
       $params[] = $query->minLongitude;
     }
 
     //deleted
-    if (isset($query->includeDeleted)) {
-      if (!($query->includeDeleted)) {
-        $where[] = 'ps.' . self::SUMMARY_STATUS . '<>?';
-        $params[] = 'DELETE';
-      }
+    if (isset($query->includeDeleted) && !$query->includeDeleted) {
```
Comment (jmfee-usgs) [inline src/lib/classes/pdl/ProductIndex.class.php:1866]: same as above, `isset` is always true

File: src/lib/classes/pdl/ProductIndex.class.php (hunk 1 above)
Comment (jmfee-usgs) [reply src/lib/classes/pdl/ProductIndex.class.php]: It is set in the class: https://github.com/usgs/earthquake-event-ws/blob/30802bdd91f65d5dc5bb2b0562a6a4f791fe739c/src/lib/classes/pdl/ProductQuery.class.php#L24

File: src/lib/classes/pdl/ProductIndex.class.php (hunk 3 above)
Comment (Capprin) [reply src/lib/classes/pdl/ProductIndex.class.php]: agreed, didn't check the query object

File: src/lib/classes/pdl/ProductIndex.class.php (hunk 4 above)
Comment (Capprin) [reply src/lib/classes/pdl/ProductIndex.class.php]: agreed, didn't check the query object

**Pull request merged**